jupyter_scheduler/scheduler.py

```diff
@@ -709,8 +709,42 @@ def get_staging_paths(self, model: Union[DescribeJob, DescribeJobDefinition]) ->
 
         output_format = "tar.gz"
         filename = create_output_filename(model.input_filename, model.create_time, output_format)
-        staging_paths[output_format] = os.path.join(self.staging_path, model.job_id, filename)
-        staging_paths["input"] = os.path.join(self.staging_path, model.job_id, model.input_filename)
+        staging_paths[output_format] = os.path.join(self.staging_path, id, filename)
+        staging_paths["input"] = os.path.join(self.staging_path, id, model.input_filename)
+
+        return staging_paths
+
+
+class AllFilesArchivingScheduler(Scheduler):
+    """Scheduler that captures all files in output directory in an archive."""
```

Comment:
This class appears to be logically identical to `ArchivingScheduler`, with two exceptions:

1. The default `execution_manager_class` is set to `AllFilesArchivingExecutionManager`

2. Minor readability changes right before the `return` statement.

I would instead apply the readability changes to `ArchivingScheduler`, have `AllFilesArchivingScheduler` inherit from that, and just override the trait's default by leaving the definition there. Then you can delete the duplicate `get_staging_paths()` implementation.

If I'm wrong about the two methods being logically identical, please see if you can use inheritance to minimize code duplication.